csr_check_test.go

```diff
@@ -1024,16 +1023,19 @@ func TestAuthorizeCSR(t *testing.T) {
 	}
 
 	for name, tc := range tests {
-		parsedCSR, err := csr.ParseCSR(&certificatesv1beta1.CertificateSigningRequest{
+		parsedCSR, err := parseCSR(&certificatesv1beta1.CertificateSigningRequest{
 			Spec: certificatesv1beta1.CertificateSigningRequestSpec{
 				Request: []byte(tc.csr),
 			},
 		})
 		if err != nil {
 			t.Fatalf("%s: error parsing test input csr %v", name, err)
 		}
-		if result := authorizeCSR(tc.machineList, tc.request, parsedCSR); result != tc.expected {
-			t.Fatalf("%s: expected %v, got %v", name, tc.expected, result)
+		err = authorizeCSR(tc.machineList, tc.request, parsedCSR)
+		if tc.expected {
+			if err != nil {
+				t.Fatalf("%v: error authorizing CSR", err)
+			}
```

Comment:
probably a better way to fix these, but i made a quick fix here to get e2e-aws rolling

Reply:
This doesn't look right, our check should be
```
expected == false
     if err == nil, fatal
else
expected == true
     if err != nil, fatal
```